includes/Context.php

```diff
@@ -357,14 +357,32 @@ public function get_amp_mode() {
 			return false;
 		}
 
-		$exposes_support_mode = method_exists( 'AMP_Theme_Support', 'get_support_mode' )
-			&& defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
+		$amp_plugin_version       = AMP__VERSION;
+		$amp_exposes_support_mode = defined( 'AMP_Theme_Support::STANDARD_MODE_SLUG' )
 			&& defined( 'AMP_Theme_Support::TRANSITIONAL_MODE_SLUG' )
 			&& defined( 'AMP_Theme_Support::READER_MODE_SLUG' );
 
+		if ( strpos( $amp_plugin_version, '-' ) !== false ) {
+			$amp_plugin_version = explode( '-', $amp_plugin_version )[0];
+		}
+
+		$amp_plugin_version_2_or_higher = version_compare( $amp_plugin_version, '2.0.0', '>=' );
```

Comment:
The `AMP__VERSION` constant _may_ not be defined. We should be super conservative and add existence checks for everything in logic like this which basically relies 100% on another plugin that we don't have control over, since it must never cause a fatal error.

I suggest we wrap the whole `$amp_plugin_version_2_or_higher` logic into a `defined( 'AMP__VERSION' )` if clause, and for the else just set it to `false` (_maybe_ a very old version of the plugin didn't have that constant).